packages/core/src/application.ts

```diff
@@ -135,6 +135,14 @@ export class Application extends Context {
     request: ServerRequest,
     response: ServerResponse,
   ) {
+    // allow CORS support for all endpoints so that users
+    // can test with online SwaggerUI instance
+    if (process.env.ENABLE_CORS) {
```

Comment:
-1 for depending on environment variables.

If you want a feature flag to controller this behaviour, then use app-level configuration, e.g. `app.getSync('allow-cors-for-api-spec')` here. Then it's up to the applications (or loopback-boot) to wire this binding to the value of `process.env.ENABLE_CORS`.

IMO, CORS should be always enabled for these endpoints.

Reply:
@bajtos Thank you for the insight. I think that can be done later, and I am fine for leaving them always enabled for now.